libheif/heif_encoder_aom.cc

```diff
@@ -95,12 +95,9 @@ void encoder_struct_aom::add_custom_option(const custom_option& p)
 {
   // if there is already a parameter of that name, remove it from list
 
-  for (size_t i = 0; i < custom_options.size(); i++) {
-    if (custom_options[i].name == p.name) {
-      for (size_t k = i + 1; k < custom_options.size(); k++) {
-        custom_options[k - 1] = custom_options[k];
-      }
-      custom_options.pop_back();
+  for (auto iter = custom_options.begin(); iter != custom_options.end(); ++iter) {
+    if (iter->name == p.name) {
+      custom_options.erase(iter);
```

Comment:
Dirk: If the order of elements in the std::vector doesn't matter, the following alternative may be more efficient:

```
  for (size_t i = 0; i < custom_options.size(); i++) {
    if (custom_options[i].name == p.name) {
      std::swap(custom_options[i], custom_options.back());
      custom_options.pop_back();
      break;
    }
  }
```

Reply:
Efficiency doesn't matter here. I'd prefer to make sure that the options are in the correct order as sometimes, this is important. E.g. a preset option followed by an option that overwrites a parameter.